src/server/services/helpers/fetch/fetch-all-proxy-rules.js

```diff
@@ -0,0 +1,25 @@
+import { config } from '~/src/config/config.js'
+import { fetcher } from '~/src/server/common/helpers/fetch/fetcher.js'
+import { createLogger } from '~/src/server/common/helpers/logging/logger.js'
+
+const logger = createLogger()
+
+export async function fetchEnvironmentProxyRules(serviceName, environment) {
+  try {
+    const endpoint =
+      config.get('portalBackendUrl') +
+      `/squid-proxy-config/${serviceName}/${environment}/`
+    const { data } = await fetcher(endpoint)
+
+    return data
+  } catch (error) {
+    if (error.output?.statusCode === 404) {
+      logger.info(
+        'Proxy rules not found, service probably not set it up on this environment'
+      )
+      return {}
+    }
+    logger.warn(error, 'Looking up Proxy rules error')
+    return {}
```

Comment:
I wouldn't silence all errors here. It looks like you just want to silence `404` as its an "expected" response. All other errors I would re-throw, like - https://github.com/DEFRA/cdp-portal-frontend/blob/main/src/server/common/helpers/fetch/fetcher.js#L36-L38